src/lib/theme/theme.service.ts

```diff
@@ -62,6 +61,17 @@ export class VantageThemeService {
 
     // apply theme on storage or media query change
     merge(storageObserver, mediaObserver).subscribe((theme: VantageTheme) => this.applyTheme(theme));
+
+    // account for cached navigation
+    // needed for Firefox BFCache
+    window.addEventListener('pageshow', (pageTransition: PageTransitionEvent) => {
+      const localStorageDiffersActiveTheme: boolean =
+        this.localStorageTheme() && this.localStorageTheme() !== this._activeTheme;
+
+      if (pageTransition.persisted && localStorageDiffersActiveTheme) {
+        this.applyTheme(this.localStorageTheme() || this.checkOSPreference());
```

Comment:
Would checkOSPreference ever be used here?

Reply:
The of localStorageDiffersActiveTheme would say no.

Reply:
Agree forgot to cleanup.